### DISCUSSION TITLE
Why my CNN IS NOT LEARNING.Couldn't able to use CNNs

### DISCUSSION THREAD

**Discussion by msarathrahul** [mrdbourke/pytorch-deep-learning#479] [General]
import torch
from torch import nn
from torch.utils.data import DataLoader,Dataset
import torchvision
from torchvision.transforms import ToTensor
from torchvision import datasets,models
train_data = datasets.FashionMNIST(root = 'data',train = True,transform=ToTensor(),download=True)
test_data = datasets.FashionMNIST(root = 'data',train = False,transform=ToTensor(),download=True)
train_batches = DataLoader(dataset=train_data,batch_size=64,shuffle=True)
test_batches = DataLoader(dataset=test_data,batch_size=64)
class ClassificationModel(nn.Module):
    
    def __init__(self):
        
        super().__init__()
        
        self.conv_b1 = nn.Sequential(nn.Conv2d(in_channels=1,out_channels=10,
                                              kernel_size=3,stride=1,padding=1),
                                    nn.ReLU(),
                                    nn.Conv2d(in_channels=10,out_channels=10,
                                             kernel_size=3,stride=1,padding=1),
                                    nn.ReLU(),
                                    nn.MaxPool2d(kernel_size=2))
        self.conv_b2 = nn.Sequential(nn.Conv2d(in_channels=10,out_channels=10,
                                              kernel_size=3,stride=1,padding=1),
                                    nn.ReLU(),
                                    nn.Conv2d(in_channels=10,out_channels=10,
                                             kernel_size=3,stride=1,padding=1),
                                    nn.ReLU(),
                                    nn.MaxPool2d(kernel_size=2))
        
        self.classifier = nn.Sequential(nn.Flatten(),
                                       nn.Linear(in_features=10*7*7,out_features=10))
        
        
    def forward(self,X):
        X = self.conv_b1(X)
        X = self.conv_b2(X)
        X = self.classifier(X)
        return X
torch.manual_seed(42)
model = ClassificationModel()
lossfn = nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(params=model.parameters(),lr = 0.1)
epochs = 3

for epoch in range(epochs):
    
    train_loss = 0
    for num,batch in enumerate(train_batches,start=1):
        
        model.train()
        
        Images,labels = batch
        
        logits = model(Images)
        
        loss = lossfn(logits,labels)
        
        train_loss += loss.item()
        
        optimizer.zero_grad()
        
        loss.backward()
        
        optimizer.step()
        
    print(train_loss/num)

THE TRAIN LOSS IS AROUND 2.3, FOR ALL THE EPOCHS.
PLEASE HELP!!

**Comment by IIsameerII** [comment]
I faced a similar issue. changing the lr (learning rate) from 0.1 to 0.01 might solve the issue.

**Reply by msarathrahul** [reply]
Thanks, IIsameerII. I figured it after posting the question!!
Thanks for taking time and going though my question.
